In [8]:
#Code to calculate the thickness when the paper is folded 43 times

THICKNESS = 0.00008

folded_thickness = THICKNESS * 2 ** 43

print("Thickness: {} meters".format(folded_thickness))

Thickness: 703687441.77664 meters


## Problem 1
How many times to fold paper to exceed the height of Mt.Fuji
t(n) = 3776 m t(0) = 0.00008 m

In [5]:
#folded paper to exceed the height of Mt.Fuji
import math # library to use logarithm
n = (math.log(3776) - math.log(0.00008)) / math.log(2)
num_of_folds = math.ceil(n) 
print("the number of folds is "+ str(num_of_folds)) 
    
    

the number of folds is 26


## [Problem 2] 
Function corresponding to arbitrary thickness.


In [9]:
##formulae for calculating distance to the non nearest star 4.0175*(pow(10, 16))

def n_folds_to_reach_height(height, thickness):
    # calculate no of folds given height and thickness  
    n = (math.log(height) - math.log(thickness)) / math.log(2) 
    
    no_of_folds = math.ceil(n) 
    # function to round up  
    print("The minimum number of folds required to exceed the height is:",no_of_folds)
    
#height to nearest star 


n_folds_to_reach_height((4.0175*(pow(10, 16))), 0.00008)

The minimum number of folds required to exceed the height is: 69


### The function takes height(meters) and thickness of paper before fold(meters), and outputs the length of paper required in meters

In [1]:
import math
def paper_length_required(height, thickness):# function to get length takes height and thickness as arguments
    print("The length is:")
    def n_folds_to_reach_height():# inner function that calculates number of folds
        n = (math.log(height) - math.log(thickness)) / math.log(2)
        no_of_folds = math.ceil(n)
        length = ((math.pi*thickness)/6)*((pow(2,no_of_folds)) + 4 )*((pow(2, no_of_folds)) - 1)# formula to calculate length of paper required
        print(length,":meters")
    return n_folds_to_reach_height()# returns the output of inner function to the outer function
# distance to moon = 384400000 meters
# paper thickness = 0.00008 meters
paper_length_required(384400000, 0.00008 )

The length is:
3.240919444170781e+21 :meters


# [Problem 3] 

Required paper length


In [59]:
import math
def paper_length_required(height, thickness):
    print("The length is:")
    def no_folds_to_reach_height():
        n = (math.log(height) - math.log(thickness)) / math.log(2)
        no_of_folds = math.ceil(n)
        length = ((math.pi*thickness)/6)*((pow(2,no_of_folds)) + 4 )*((pow(2, no_of_folds)) - 1)# formula to calculate length of paper required
        print(length,":meters")
    return no_folds_to_reach_height()# returns the output of inner function to the outer function
# distance to proxima centauri = 4.0175*10^16 meters
# paper thickness = 0.00008 meters
paper_length_required((4.0175*(pow(10,16))), 0.00008 )

The length is:
1.4595803601100348e+37 :meters


The length of paper required to fold thickness equal to distance of nearest star proxima centauri is (1.46 * 10^37) meters